aionetbox/api.py

```diff
@@ -0,0 +1,302 @@
+        self.loop = loop or asyncio.get_event_loop()
+
+    async def request(self, method, url, query_params=None, headers=None, body=None, post_params=None, _timeout=None):
+        method = method.upper()
+        assert method in ['GET', 'HEAD', 'DELETE', 'POST', 'PUT', 'PATCH', 'OPTIONS']
+
+        if post_params and body:
+            raise ValueError(
+                'body parameter cannot be used with post_params parameter.'
+            )
+
+        post_params = post_params or {}
+        headers = headers or {}
+        headers.update({
+            'Authorization': 'Token {}'.format(self.api_key)
+        })
+        timeout = _timeout or 5 * 60
+
+        if 'Content-Type' not in headers:
+            headers['Content-Type'] = 'application/json'
+
+        args = {
+            'method': method,
+            'url': url,
+            'timeout': timeout,
+            'headers': headers
+        }
+
+        if query_params:
+            args['url'] += '?' + urlencode(query_params)
+
+        if method in ['POST', 'PUT', 'PATCH', 'OPTIONS', 'DELETE']:
+            if re.search('json', headers['Content-Type'], re.IGNORECASE):
+                if body is not None:
+                    body = json.dumps(body)
+                args['data'] = body
+            elif headers['Content-Type'] == 'application/x-www-form-urlencoded':  # noqa: E501
+                args['data'] = aiohttp.FormData(post_params)
+
+            # Pass a `bytes` parameter directly in the body to support
+            # other content types than Json when `body` argument is provided
+            # in serialized form
+            elif isinstance(body, bytes):
+                args['data'] = body
+            else:
+                # Cannot generate the request from given parameters
+                msg = '''Cannot prepare a request message for provided arguments. Please check that your
+                         arguments match declared content type.'''
+                raise Exception(msg)
+
+        return await self.session.request(**args)
+
+    def parse_spec(self, config):
+        operations = {}
+
+        if not config:
+            return {}
+
+        if isinstance(config, ResolvingParser):
+            config = config.specification
+
+        operations['_orig'] = config
+        for path, data in config.get('paths', {}).items():
+            for action in self._http_methods:
+                if action not in data:
+                    continue
+
+                payload = data.get(action, {})
+                tags = payload.get('tags', [])
+
+                for tag in tags:
+                    if tag not in operations:
+                        operations[tag] = {}
+
+                    rest_cfg = {
+                        'url': path,
+                        'params': data.get('parameters', []),
+                        'method': action,
+                    }
+
+                    operations[tag][payload.get('operationId')] = payload
```

Comment:
if `payload` resolves to `{}` above (L115), then this should set the key to `None` -- is that okay here? (not sure what the data expectations are, so just double checking)